Add Delete a Deployment Topic (#940)
astronomer/docs: astro/configure-deployment-resources.md

    The Airflow components of your Deployment automatically restart to apply the updated resource allocations. This action is equivalent to deploying code to your Deployment and does not impact running tasks that have 24 hours to complete before running workers are terminated. See [What happens during a code deploy](deploy-code.md#what-happens-during-a-code-deploy).

## Delete a Deployment

> When you delete a Deployment, all infrastructure resources assigned to the Deployment are deleted. After you delete a Deployment, the Deployment Kubernetes namespace and the Deployment metadata database in your data plane are retained for 30 days. Deleted Deployments can't be restored. If you accidentally delete a Deployment, contact [Astronomer support](https://support.astronomer.io).

Reviewer: ```suggestion
When you delete a Deployment, all infrastructure resources assigned to the Deployment are deleted from your data plane. All of these resources are deleted immediately except for the Deployment's Kubernetes namespace and metadata database, which are retained for 30 days. Deleted Deployments can't be restored. If you accidentally delete a Deployment, contact [Astronomer support](https://support.astronomer.io).
```
@paolaperaza A couple of clarifying product questions: 
- What's the implication for the user that the namespace and metadata db are retained for 30 days? What does this mean for them if they can't restore a Deployment? 
- What will Astronomer support do for accidentally deleted Deployments if they can't be restored?

Author: @jwitz we should be avoiding the possessive form for inanimate objects (Deployment's) where possible. Inanimate objects or abstract ideas cannot possess anything. I understand that this is a contentious issue and sometimes it can't be avoided. However, in this instance, it can be avoided.

The _[Google developer documentation style guide](https://developers.google.com/style/possessives?hl=en)_ provides this recommendation:

_Don't form a possessive from a feature name, product name, or trademark, regardless of who owns it. Instead, use the name as a modifier or rewrite to use a word like of to indicate the relationship._

As _Deployment_ is a feature name, the possessive form shouldn't be used.

> When you delete a Deployment, all infrastructure resources assigned to the Deployment are deleted from your data plane. The Deployment Kubernetes namespace and the Deployment metadata database are retained for 30 days after a Deployment is deleted. Deleted Deployments can't be restored. If you accidentally delete a Deployment, contact [Astronomer support](https://support.astronomer.io).

Reviewer 2: ```suggestion
When you delete a Deployment, all infrastructure resources assigned to the Deployment are immediately deleted from your data plane. The Kubernetes namespace and metadata database for the Deployment are retained for 30 days after it is deleted. Deleted Deployments can't be restored. If you accidentally delete a Deployment, contact [Astronomer support](https://support.astronomer.io).
```
